rotkehlchen/chain/bitcoin/__init__.py

```diff
@@ -13,53 +13,98 @@ def _have_bc1_accounts(accounts: List[BTCAddress]) -> bool:
+    - KeyError if got problems with reading the returned json
+    """
+    balances = {}
+    for account in accounts:
+        url = f'https://mempool.space/api/address/{account}'
+        response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
+        stats = response_data['chain_stats']
+        balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
+        balances[account] = satoshis_to_btc(balance)
+    return balances
+
 
+def _query_blockchain_info(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
+    """Queries balances from blockchain.info
     May raise:
-    - RemotError if there is a problem querying blockchain.info or blockstream
+    - RemoteError if got problems with querying the API
+    - KeyError if got problems with reading the returned json
     """
-    source = 'blockchain.info'
     balances: Dict[BTCAddress, FVal] = {}
-    try:
-        if _have_bc1_accounts(accounts):
-            # if 1 account is bech32 we have to query blockstream. blockchaininfo won't work
-            source = 'blockstream'
-            balances = {}
-            for account in accounts:
-                url = f'https://blockstream.info/api/address/{account}'
-                response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
-                stats = response_data['chain_stats']
-                balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
-                balances[account] = satoshis_to_btc(balance)
-        else:
-            # split the list of accounts into sublists of 80 addresses per list to overcome:
-            # https://github.com/rotki/rotki/issues/3037
-            accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
-            for accounts_chunk in accounts_chunks:
-                params = '|'.join(accounts_chunk)
-                btc_resp = request_get_dict(
-                    url=f'https://blockchain.info/multiaddr?active={params}',
-                    handle_429=True,
-                    # If we get a 429 then their docs suggest 10 seconds
-                    # https://blockchain.info/q
-                    backoff_in_seconds=10,
-                )
-                for entry in btc_resp['addresses']:
-                    balances[entry['address']] = satoshis_to_btc(FVal(entry['final_balance']))
-    except (
+    accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
+    for accounts_chunk in accounts_chunks:
+        params = '|'.join(accounts_chunk)
+        btc_resp = request_get_dict(
+            url=f'https://blockchain.info/multiaddr?active={params}',
+            handle_429=True,
+            # If we get a 429 then their docs suggest 10 seconds
+            # https://blockchain.info/q
+            backoff_in_seconds=10,
+        )
+        for entry in btc_resp['addresses']:
+            balances[entry['address']] = satoshis_to_btc(FVal(entry['final_balance']))
```

Comment:
Do not initialize `FVal()` directly from the constructor if you read data coming from the outside. It may fail badly if the data is not correct. Use `deserialize_fval` and follow the same advice I gave for the previous function.